src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -200,76 +188,73 @@ public boolean isCommentNodesRequired() {
-                comment = commentList.get(commentList.size() - 1);
-                lineBefore = line.substring(0, comment.getStartColNo());
-
-                // do not check comment which doesn't end line
-                if (comment.getText().length == 1
-                        && !CommonUtil.isBlank(line
-                            .substring(comment.getEndColNo() + 1))) {
-                    continue;
-                }
-            }
-            if (!format.matcher(lineBefore).find()
-                && !isLegalComment(comment)) {
-                illegalLines.put(lineNo, comment.getStartColNo());
+                isCommentLegal = true;
             }
         }
+
+        if (!isCommentLegal && !isBeforeCommentLegal(ast)) {
+            log(ast, MSG_KEY);
+        }
     }
 
     /**
-     * Checks if given comment is legal (single-line and matches to the
-     * pattern).
+     * Check if commentContent matches the legalComment pattern.
      *
-     * @param comment comment to check.
-     * @return true if the comment if legal.
+     * @param commentContent the content of a comment
+     * @return true if commentContent matches the legalComment pattern
      */
-    private boolean isLegalComment(final TextBlock comment) {
-        final boolean legal;
+    private boolean isCommentMatchLegalPattern(String commentContent) {
+        return legalComment != null && legalComment.matcher(commentContent.trim()).find();
+    }
 
-        // multi-line comment can not be legal
-        if (legalComment == null || comment.getStartLineNo() != comment.getEndLineNo()) {
-            legal = false;
+    /**
+     * Get the text after the end of a block comment.
+     *
+     * @param commentEndAst the comment end ast
+     * @return the text after the end of a block comment
+     */
+    private String getTextAfterBlockComment(DetailAST commentEndAst) {
+        final String commentEndLine = getLine(commentEndAst.getLineNo() - 1);
+        final int commentEndColumnNo = commentEndAst.getColumnNo();
+        final String afterComment;
+        if (commentEndLine.charAt(commentEndColumnNo) == '*') {
+            // commentEndColumnNo is pointing at the asterisk of */
+            afterComment = commentEndLine.substring(commentEndColumnNo + 2);
         }
         else {
-            String commentText = comment.getText()[0];
-            // remove chars which start comment
-            commentText = commentText.substring(2);
-            // if this is a C-style comment we need to remove its end
-            if (commentText.endsWith("*/")) {
-                commentText = commentText.substring(0, commentText.length() - 2);
-            }
-            commentText = commentText.trim();
-            legal = legalComment.matcher(commentText).find();
+            // commentEndColumnNo is pointing at the space right before */
+            afterComment = commentEndLine.substring(commentEndColumnNo + 1 + 2);
```

Comment:
again, some duplication.
I suggest to make it like
```
int offset = 2;
if (commentEndLine.charAt(commentEndColumnNo) != '*') {
offset = 3;
}
return commentEndLine.substring(commentEndColumnNo + offset);
```

Reply:
fixed. `offset = 3` gives error `3 is a magic number`. So I wrote `offset += 1` instead